launcher/launcher.go

```diff
@@ -1852,643 +911,615 @@ func (launcher *Launcher) getHostsFromResources(resources []string) (hosts []con
+		case "proc", "tmpfs", "sysfs":
+			if err := createMountPoint(path, mount, true); err != nil {
+				return err
+			}
+
+		case "bind":
+			stat, err := os.Stat(mount.Source)
+			if err != nil {
+				return aoserrors.Wrap(err)
+			}
+
+			if err := createMountPoint(path, mount, stat.IsDir()); err != nil {
+				return err
+			}
 		}
 	}
 
-	return false
+	return nil
 }
 
-func (launcher *Launcher) validateTTLs() {
-	launcher.Lock()
+func (launcher *Launcher) handleTTLs(ctx context.Context) {
+	for {
+		select {
+		case <-time.After(CheckTLLsPeriod):
+			launcher.Lock()
+			launcher.updateInstancesEnvVars()
+			launcher.Unlock()
 
-	if launcher.ttlTicker != nil {
-		launcher.Unlock()
-		return
+		case <-ctx.Done():
+			return
+		}
 	}
+}
 
-	launcher.ttlTicker = time.NewTicker(ttlValidatePeriod)
-	defer launcher.ttlTicker.Stop()
-
-	launcher.ttlRemoveServices = time.NewTicker(ttlRemoveServices)
-	defer launcher.ttlRemoveServices.Stop()
+func (launcher *Launcher) stopRunningInstances() error {
+	log.Debug("Stop running instances")
 
-	launcher.Unlock()
+	runningInstances, err := launcher.storage.GetRunningInstances()
+	if err != nil {
+		return aoserrors.Wrap(err)
+	}
 
-	for {
-		select {
-		case <-launcher.ttlTicker.C:
-			subjectServiceToRestart, err := launcher.envVarsProvider.validateEnvVarsTTL()
-			if err != nil {
-				log.Error("Validate env var ttl error: ", err)
-				continue
-			}
+	launcher.cacheCurrentServices(runningInstances)
 
-			launcher.restartServicesBySubjectServiceID(subjectServiceToRestart)
+	stopInstances := make([]*instanceInfo, 0, len(runningInstances))
 
-		case <-launcher.ttlRemoveServices.C:
-			if err := launcher.cleanCache(); err != nil {
-				log.Errorf("Error cleaning cache: %s", err)
-			}
+	for _, runningInstance := range runningInstances {
+		service, err := launcher.getCurrentServiceInfo(runningInstance.ServiceID)
+		if err != nil {
+			log.WithFields(
+				instanceIdentLogFields(runningInstance.InstanceIdent, nil),
+			).Errorf("Can't get service info: %v", err)
 
-		case <-launcher.ttlStopChannel:
-			return
+			continue
 		}
+
+		stopInstances = append(stopInstances, &instanceInfo{
+			InstanceInfo: runningInstance,
+			isStarted:    true,
+			service:      service,
+		})
 	}
+
+	launcher.stopInstances(stopInstances)
+
+	return nil
 }
 
-func (launcher *Launcher) checkServiceHealth(serviceName string) (err error) {
-	subSet := launcher.systemd.NewSubscriptionSet()
+func (launcher *Launcher) removeOutdatedInstances() error {
+	log.Debug("Remove outdated instances")
 
-	subSet.Add(serviceName)
+	instances, err := launcher.storage.GetAllInstances()
+	if err != nil {
+		return aoserrors.Wrap(err)
+	}
 
-	evChan, errChan := subSet.Subscribe()
+	launcher.cacheCurrentServices(instances)
 
-	var curretActiveStatus string
+	for _, instance := range instances {
+		if _, serviceErr := launcher.getCurrentServiceInfo(instance.ServiceID); serviceErr != nil {
+			if errors.Is(serviceErr, servicemanager.ErrNotExist) {
```

Comment:
I think that in this case the check will not work, since the function  `getCurrentServiceInfo` returns a different error